bin/cli.js

```diff
@@ -1,17 +1,242 @@
+      if (firstLog) {
+        firstLog = false;
+        console.log(`\n${browser}:`);
+      }
+      console.log(log.slice(1, log.length - 1));
+    };
+  };
+
+  reporter.$inject = ['baseReporterDecorator'];
+  return reporter;
+};
+
+const getBrowserConfig = (conf) => {
+  const config = {
+    preprocessors: {},
+    files: [],
+    plugins: [
+      'karma-webpack',
+      { 'reporter:meta': ['type', getReporter()] }
+    ],
+    reporters: [ 'meta' ],
+    basePath: process.env.PWD,
+    port: conf.browser.port,
+    autoWatch: false,
+    singleRun: true,
+    concurrency: 1
+  };
+
+  const adapter = path.resolve('./build/adapter.js');
+  config.files.push(adapter);
+  config.preprocessors[adapter] = ['webpack'];
+
+  const nodePackages = [ 'fs', 'child_process' ];
+
+  removeNodePackages(config, ...nodePackages);
+  setKarmaBrowsers(config, ...conf.browser.browsers);
+  setKarmaLogLevel(config, conf.browser.logLevel);
+  return config;
+};
 
-walk('../test', (err, files) => {
-  metasync.for(files).each((file, next) => {
-    require(file.filePath);
-    next();
-  }).fetch(() => {
-    console.log('Tests done');
+const getEnvironment = (config, program) => {
+  if (program.node || program.browser) {
+    config.environments = [];
+    if (program.node) config.environments.push('node');
+    if (program.browser) config.environments.push('browser');
+  }
+  if (!config.environments || config.environments.length === 0)
+    config.environments = [ 'node' ];
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  cliOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+  getEnvironment(config, program);
+
+  config.files = config.files ?
+    config.files.concat(program.args) : program.args;
+  config.files = config.files.length ? config.files : [TEST_DIR];
+  config.files = loadFiles(config.files);
+
+  config.packages = config.packages ?
+    config.packages.concat(program.requiredPackages) : program.requiredPackages;
+  config.packages = rmDuplicates(config.packages);
+
+  if (config.environments.indexOf('browser') >= 0) {
+    config.browser = config.browser || {};
+    config.browser.browsers = config.browser.browsers ?
+      config.browser.browsers.concat(program.browsers) : program.browsers;
+    config.browser.browsers = rmDuplicates(config.browser.browsers);
+
+    config.browser.logLevel = program.browserLog || config.browser.logLevel;
+    config.browser.port = +program.browserPort || config.browser.port;
+    config.browser = getBrowserConfig(config);
+  }
+  return config;
+};
+
+const runNode = (config, cb) => {
+  // TODO(SemenchenkoVitaliy): remove after all
+  // tests migrate to metatests 0.2.0
+  global.api = { metatests: require('..') };
```

Comment:
Actually, we can. I just left it there according to https://github.com/metarhia/metatests/pull/75#discussion_r214552356. And question about tests that should be run only in node or only in browser. Should we just have 2 separate configs or add some options which will run some files based on environment?

Reply:
There is no reason for this hack, because this feature will re released in `0.2.1`, thus any package that will install it should be already upgadated to `0.2.0`.

As per your question: I guess having two separate options for browser and Node.js is a good idea. WDYT @lundibundi?

Reply:
@lundibundi this should be defined in a config because It's not a good practice to force a user to use some specific folder we defined.